modules/gdscript/gdscript_parser.cpp

```diff
@@ -3533,6 +3533,11 @@ void GDScriptParser::_parse_class(ClassNode *p_class) {
 					return;
 				}
 
+				if (p_class->classname_used && ScriptServer::is_global_class(p_class->name)) {
```

Comment:
This check already exists above (line 3526), no?

Reply:
Isn't that checking for duplicate singletons? It doesn't do any checks for types declared with `class_name`.

Reply:
Wouldn't this check error even on scripts using class_name without being singletons?

Reply:
I will check that this afternoon, but wouldn't it only act on singletons due to the following line?
```
ScriptServer::is_global_class(p_class->name)
```

Reply:
I'm pretty sure singletons/autoloads are not handled by `is_global_class`. More like, `is_global_class` should handle `class_name`-s

Reply:
Oh... but what is `p_class->classname_used` used for then?

Reply:
`p_class->classname_used` is set to true whenever `class_name` (the keyword) was used in the class.
To check if there is a singleton with that name, you would have to check `ProjectSettings`. Specifically, search for `"autoload/"` in the code.